### PR TITLE
Setup appsettings.json environment config files

### PR DESCRIPTION
**Pull request by jonkjetiloye**
## Description
- All non-secret config values (API endpoints etc.) are now setup directly in repo in different environment appsettings.json files (appsettings.AT22.json, appsettings.TT02.json, appsettings.Prod.json etc.)
- Renamed some values for clearity and removed some unused appsettings values

## Related Issue(s)
- #298

## Verification
- [x] **Your** code builds clean without any errors or warnings
- [x] Manual testing done (required)
- [ ] Relevant automated test added (if you find this hard, leave it and we'll help out)
- [x] All tests run green

## Documentation
- [ ] User documentation is updated with a separate linked PR in [altinn-studio-docs.](https://github.com/Altinn/altinn-studio-docs) (if applicable)

### CODE REVIEW

File: backend/src/Altinn.AccessManagement.UI/Altinn.AccessManagement.UI/appsettings.AT23.json [hunk 1]
```diff
@@ -0,0 +1,14 @@
+{
+  "PlatformSettings": {
```
Comment (jonkjetiloye) [inline backend/src/Altinn.AccessManagement.UI/Altinn.AccessManagement.UI/appsettings.AT23.json:2]: missing at21 urls in at23 config

File: backend/src/Altinn.AccessManagement.UI/Altinn.AccessManagement.UI.Core/Configuration/GeneralSettings.cs [hunk 2]
```diff
@@ -5,6 +5,11 @@ namespace Altinn.AccessManagement.UI.Core.Configuration
     /// </summary>
     public class GeneralSettings
     {
+        /// <summary>
```
[the remark this answers is not in the document]
Comment (acn-dgopa) [reply backend/src/Altinn.AccessManagement.UI/Altinn.AccessManagement.UI.Core/Configuration/GeneralSettings.cs:8]: I see that this is used differently now

File: backend/src/Altinn.AccessManagement.UI/Altinn.AccessManagement.UI/Controllers/HomeController.cs [hunk 3]
```diff
@@ -77,13 +78,13 @@ public async Task<IActionResult> Index()
...
             await SetLanguageCookie();
-            
+
             if (ShouldShowAppView())
             {
                 return View();
             }
 
-            string goToUrl = HttpUtility.UrlEncode($"{_platformSettings.AltinnPlatformBaseUrl}{Request.Path}");
+            string goToUrl = HttpUtility.UrlEncode($"{_generalSettings.FrontendBaseUrl}{Request.Path}");
```
[the remark this answers is not in the document]
Comment (acn-dgopa) [reply backend/src/Altinn.AccessManagement.UI/Altinn.AccessManagement.UI/Controllers/HomeController.cs:87]: I see the config value is as expected. just that its used different now.

File: backend/src/Altinn.AccessManagement.UI/Altinn.AccessManagement.UI/appsettings.AT21.json [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+    "ApiProfileEndpoint": "https://platform.at21.altinn.cloud/profile/api/v1/",
+    "ApiRegisterEndpoint": "https://platform.at21.altinn.cloud/register/api/v1/",
+    "ApiResourceRegistryEndpoint": "https://platform.at21.altinn.cloud/resourceregistry/api/v1/",
+    "OpenIdWellKnownEndpoint": "https://platform.at21.altinn.cloud/authentication/api/v1/openid/"
+  },
+  "GeneralSettings": {
+    "FrontendBaseUrl": "https://am.ui.at21.altinn.cloud/",
+    "Hostname": "at21.altinn.cloud"
+  }
+}
```
Comment (acn-dgopa) [inline backend/src/Altinn.AccessManagement.UI/Altinn.AccessManagement.UI/appsettings.AT21.json:14]: We also need cache config, client settings, keyvaultsettings

Comment (acn-dgopa) [reply backend/src/Altinn.AccessManagement.UI/Altinn.AccessManagement.UI/appsettings.AT21.json:14]: I guess we need to save the clientsettings and keyvaultsettings as environment secret

**Review by acn-dgopa — APPROVED** [review]

File: backend/src/Altinn.AccessManagement.UI/Altinn.AccessManagement.UI.Integration/Clients/DelegationsClient.cs [hunk 5]
```diff
@@ -28,7 +22,6 @@ public class DelegationsClient : IDelegationsClient
         private readonly HttpClient _client;
         private readonly IHttpContextAccessor _httpContextAccessor;
         private readonly PlatformSettings _platformSettings;
-        private readonly IAccessTokenGenerator _accessTokenGenerator;
         private readonly JsonSerializerOptions _serializerOptions = new JsonSerializerOptions { PropertyNameCaseInsensitive = true };
```
[the remark this answers is not in the document]
Comment (jonkjetiloye) [reply backend/src/Altinn.AccessManagement.UI/Altinn.AccessManagement.UI.Integration/Clients/DelegationsClient.cs:25]: it's already a part of the constructor

File: backend/src/Altinn.AccessManagement.UI/Altinn.AccessManagement.UI/appsettings.AT21.json (hunk 4 above)
Comment (jonkjetiloye) [reply backend/src/Altinn.AccessManagement.UI/Altinn.AccessManagement.UI/appsettings.AT21.json:14]: cache config will for now just use the defaults in appsettings.json. 
No need to introduce environment specific values until we actually have different values for different environments.

yes, the keyvault and client settings should come from environment secrets

**Pull request merged**